src/test/java/core/basesyntax/RegistrationServiceTest.java

```diff
@@ -0,0 +1,112 @@
+
+import core.basesyntax.dao.StorageDaoImpl;
+import core.basesyntax.model.User;
+import core.basesyntax.service.RegistrationServiceImpl;
+import java.util.List;
+import org.junit.jupiter.api.AfterAll;
+import org.junit.jupiter.api.BeforeAll;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.TestInstance;
+
+@TestInstance(PER_CLASS)
+public class RegistrationServiceTest {
+    private RegistrationServiceImpl registrationService;
+    private StorageDaoImpl dao;
+    private List<User> validUsers;
+
+    @BeforeAll
+    public void setup() {
+        registrationService = new RegistrationServiceImpl();
+        dao = new StorageDaoImpl();
+        validUsers = List.of(new User("James", "qwerty",20),
+                        new User("Anna", "password", 30),
+                        new User("Alice", "123456", 40));
+    }
+
+    @AfterAll
+    public void close(){
+    }
+
+    @Test
+    public void register_nullUser_throwsNullPointer() {
+        var exception = assertThrows(NullPointerException.class,
+                () -> registrationService.register(null));
+        assertEquals("User required", exception.getMessage());
+    }
+
+    @Test
+    public void register_userNullProperties_throwsNullPointer() {
+        User userWithNullLogin = new User(null, "12344321", 20);
+        User userWithNullPassword = new User("Sam", null, 20);
+        User userWithNullAge = new User("Mickey", "12344321", null);
+        var nullLoginException = assertThrows(NullPointerException.class,
+                () -> registrationService.register(userWithNullLogin));
+        var nullPasswordException = assertThrows(NullPointerException.class,
+                () -> registrationService.register(userWithNullPassword));
+        var nullAgeException = assertThrows(NullPointerException.class,
+                () -> registrationService.register(userWithNullAge));
+        assertEquals("Login required", nullLoginException.getMessage());
+        assertEquals("Password required", nullPasswordException.getMessage());
+        assertEquals("Age required", nullAgeException.getMessage());
+    }
+
+    @Test
+    public void register_userEmptyLogin_throwsRuntimeException() {
+        User userWithEmptyLogin = new User("", "12344321", 20);
+        User userWithBlankLogin = new User("   ", "12344321", 20);
+        var exceptionEmptyLogin = assertThrows(RuntimeException.class,
+                () -> registrationService.register(userWithEmptyLogin));
+        var exceptionBlankLogin = assertThrows(RuntimeException.class,
+                () -> registrationService.register(userWithBlankLogin));
+        assertEquals("Login is empty", exceptionEmptyLogin.getMessage());
+        assertEquals("Login is empty", exceptionBlankLogin.getMessage());
+    }
+
+    @Test
+    public void register_userWithExistingLogin_throwsRuntimeException() {
+        User user = new User("User123", "qwerty", 30);
+        registrationService.register(user);
```

Comment:
lets add user to storage directly, in case the register() method does not work correctly